src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -2154,6 +2167,10 @@ bool policy::CacheManager::SetIsPredata(const std::string& app_id) {
   if (IsApplicationRepresented(app_id)) {
     pt_->policy_table.app_policies_section.apps[app_id].set_to_string(
         kPreDataConsentId);
+
+    pt_->policy_table.app_policies_section.apps[app_id].heart_beat_timeout_ms =
```

Comment:
we just set the entry to a string, not an object. this line is trying to assign properties on a string and should be removed.

`pt_->policy_table.app_policies_section.apps[app_id]`:
```
"app_policies": {
  "app_id": "pre_DataConsent"
}
```
we don't need to set properties on `app_policies.app_id` as it is a reference to `app_policies.pre_DataConsent` where the correct heartbeat timeout ms should live.

Reply:
@iCollin in this example we just mark that entry as a reference to preData entity and assign value of `preData.heart_beat_timeout_ms` to `app_id.heart_beat_timeout_ms`, however preData is still the initial source of that value. Looks like there was no another place where SDL pulls that value from preData.
When snapshot is generated, this `app_id` will be properly collapsed into record you mentioned above.